subscription/subscription_worker.py

```diff
@@ -0,0 +1,137 @@
+from sys import stdout
+from botocore.exceptions import ClientError
+
+AWS_REGION = os.getenv("AWS_REGION")
+QUEUE_URL = os.getenv("QUEUE_URL")
+DEAD_LETTER_QUEUE_URL = os.getenv("DEAD_LETTER_QUEUE_URL")
+SUB_DEAD_LETTER_QUEUE_URL = os.getenv("SUB_DEAD_LETTER_QUEUE_URL")
+LONG_POLL_TIME = os.getenv("LONG_POLL_TIME")
+SNS_NAME = os.getenv("SNS_NAME")
+
+class sns:
+    """Encapsulates AWS SNS topics."""
+
+    def __init__(self, sns_resource):
+        """:param sns_resource: A Boto3 AWS SNS resource."""
+        self.sns_resource = sns_resource
+
+    def create_topic(self, topic_name):
+        """The topic is already created, but this is best practice to get the topic client
+        from AWS."""
+        try:
+            topic = self.sns_resource.create_topic(Name=topic_name)
+        except ClientError as error:
+            print("Count not get the topic ARN: {error}.")
+            stdout.flush()
+            raise error
+        else:
+            return topic
+
+    @staticmethod
+    def publish_message(topic, message):
+        """ Publishes a message with attributes to the CMR external topic. Subscriptions
+        can be filtered based on the message attributes. """
+        message_body_str = message["Body"]
+        message_body = json.loads(message_body_str)
+        message_subject = message_body["Subject"]
+        message_attributes = message_body["MessageAttributes"]
+        message_message = message_body["Message"]
+        try:
+            if message_attributes:
+                att_dict = {}
+                for key in message_attributes.keys():
+                    att_dict[key] = {"DataType": "String", "StringValue": message_attributes[key]["Value"]}
+                response = topic.publish(Subject=message_subject, Message=message_message, MessageAttributes=att_dict)
+            else:    
+                response = topic.publish(Subject=message_subject, Message=message_message)
+        except ClientError as error:
+            print(f"Could not publish message to topic {topic}. {error}")
+            stdout.flush()
+            raise error
+        else:
+            return response
+
+def receive_message(sqs_client, queue_url):
+    response = sqs_client.receive_message(
+        QueueUrl=queue_url,
+        MaxNumberOfMessages=1,
+        # Long Polling
+        WaitTimeSeconds=(int (LONG_POLL_TIME)))
+
+    if (len(response.get('Messages', [])) > 0):
+        print(f"Number of messages received: {len(response.get('Messages', []))}")
+        stdout.flush()
+    return response
+
+def delete_message(sqs_client, queue_url, receipt_handle):
+    response = sqs_client.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
+
+def delete_messages(sqs_client, queue_url, messages):
+    for message in messages.get("Messages", []):
+        receipt_handle = message['ReceiptHandle']
+        delete_message(sqs_client=sqs_client, queue_url=queue_url, receipt_handle=receipt_handle)
+
+def check_acls(message):
```

Comment:
unused message param -- can remove

Reply:
what is the purpose of this func? It doesn't seem like it does anything at the moment

Reply:
It is/was a stub for checking ACLs to see if we need to forward the message on or if we needed to withhold the message because of ACLs.  Since we now do not have to do the ACL work at the moment, I will remove this function.

Reply:
I have removed this function.  We will add it back in, in the future when we will do the ACL work.